lm75bd/lm75bd.c

```diff
@@ -1,74 +1,90 @@
 #include "lm75bd.h"
-#include "i2c_io.h"
-#include "errors.h"
 
+#include <math.h>
 #include <stdint.h>
 #include <string.h>
-#include <math.h>
+
+#include "errors.h"
+#include "i2c_io.h"
 
 /* LM75BD Registers (p.8) */
-#define LM75BD_REG_CONF 0x01U  /* Configuration Register (R/W) */
+#define LM75BD_REG_CONF 0x01U /* Configuration Register (R/W) */
+#define LM75BD_REG_TEMP 0x00U /* Temperature Register */
 
 error_code_t lm75bdInit(lm75bd_config_t *config) {
-  error_code_t errCode;
+    error_code_t errCode;
 
-  if (config == NULL) return ERR_CODE_INVALID_ARG;
+    if (config == NULL) return ERR_CODE_INVALID_ARG;
 
-  errCode = writeConfigLM75BD(config->devAddr, config->osFaultQueueSize, config->osPolarity,
-                                         config->osOperationMode, config->devOperationMode);
-  
-  if (errCode != ERR_CODE_SUCCESS) return errCode;
+    errCode = writeConfigLM75BD(config->devAddr, config->osFaultQueueSize,
+                                config->osPolarity, config->osOperationMode,
+                                config->devOperationMode);
 
-  // Assume that the overtemperature and hysteresis thresholds are already set
-  // Hysteresis: 75 degrees Celsius
-  // Overtemperature: 80 degrees Celsius
+    if (errCode != ERR_CODE_SUCCESS) return errCode;
 
-  return ERR_CODE_SUCCESS;
+    // Assume that the overtemperature and hysteresis thresholds are already set
+    // Hysteresis: 75 degrees Celsius
+    // Overtemperature: 80 degrees Celsius
+
+    return ERR_CODE_SUCCESS;
 }
 
 error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
-  /* Implement this driver function */
-  
-  return ERR_CODE_SUCCESS;
+    if (temp == NULL) return ERR_CODE_INVALID_ARG;
+
+    uint8_t regTemp = LM75BD_REG_TEMP;
+    error_code_t writeErrCode = i2cSendTo(devAddr, &regTemp, 1);
+    if (writeErrCode != ERR_CODE_SUCCESS) return writeErrCode;
+
+    uint8_t readBuf[2] = {0};
+    error_code_t readErrCode = i2cReceiveFrom(devAddr, readBuf, 2);
```

Comment:
You don't need to define 2 separate error code variables. We typically define a general error code variable like `error_code_t errCode` at the top of a function and use it throughout the function